File: cgma/api/lexer.js
const { run: lexerRun } = require('../../cgmalexer');

module.exports = async (req, res) => {
    const { source_code } = req.body;
    const [tokens, errors] = lexerRun('<stdin>', source_code);

    res.status(200).json({
        tokens: tokens.map(token => ({ type: token.type, value: token.value })),
        errors: errors.map(error => error.as_string())
    });
};

File: cgma/api/lexer.py
from cgmalexer import run as lexer_run
import json

def handler(event, context):
    body = json.loads(event['body'])
    source_code = body.get('source_code', '')

    tokens, errors = lexer_run('<stdin>', source_code)

    response = {
        'tokens': [{'type': token.type, 'value': token.value} for token in tokens],
        'errors': [error.as_string() for error in errors]
    }

    return {
        'statusCode': 200,
        'body': json.dumps(response)
    }
